app/views/data_element_values/_form.html.erb

```diff
@@ -1,29 +1,89 @@
 <%= form_for @data_element_value, :html => { :class => 'form-horizontal form-model' } do |f| %>
   <div class="form-group">
-    <%= f.label :environment, :class => 'control-label col-md-2' %>
-    <div class="col-md-10">
+    <%= f.label :environment, :class => 'control-label col-md-3' %>
+    <div class="col-md-9">
       <%= f.collection_select :environment_id, @environments, :id, :name, {:class => 'text-field form-control'} %>
     </div>
   </div>
   <div class="form-group">
-    <%= f.label :variable, :class => 'control-label col-md-2' %>
-    <div class="col-md-10">
+    <%= f.label :variable, :class => 'control-label col-md-3' %>
+    <div class="col-md-9">
       <%= f.collection_select :data_element_id, @variables, :id, :key, {:class => 'text-field form-control'} %>
     </div>
   </div>
-  <div class="form-group">
-    <%= f.label :value, :class => 'control-label col-md-2' %>
-    <div class="col-md-10">
+  <div class="form-group" id="variable-text-entry">
+    <%= f.label :value, :class => 'control-label col-md-3' %>
+    <div class="col-md-9">
       <%= f.text_field :value, :class => 'text-field form-control' %>
     </div>
   </div>
+  <div class="form-group variable-control-container" id="variable-length-entry">
+    <%= f.label :random_value_length, :class => 'control-label col-md-3' %>
+    <div class="col-md-9">
+      <%= f.number_field :random_value_length,
+          :class => 'text-field form-control' %>
+    </div>
+  </div>
   <%= f.hidden_field :data_element_id, :value => @data_element_value.data_element_id %>
 
+  <div class="form-group variable-control-container">
+    <%= f.label :random_value, :class => 'control-label col-md-3' %>
+    <div class="col-md-9">
+      <%= f.check_box :random_value,
+          :class => 'text-field form-control',
+          :id => 'variable-control',
+          :checked => @data_element_value.random_value %>
```

Comment:
Here's what this looks like to me: 
<img width="496" alt="screen shot 2016-04-29 at 8 28 44 am" src="https://cloud.githubusercontent.com/assets/2657187/14910373/890c24ee-0de4-11e6-9d24-ddd2f90ab73c.png">